### PR TITLE
Fixes vending machines borking their inventory

### PR DESCRIPTION
**Pull request by Tastyfish**
- The main issue was that `RefreshParts()` added the starting inventory on top of the existing inventory every time it ran...which is bad.
- `user` was also mishandled in `refill_inventory()`.

### CODE REVIEW

**Comment by KasparoVy** [comment]
Does this mean that deconstructing and reconstructing vending machines won't restock them?

File: code/game/machinery/vending.dm [hunk 1]
```diff
@@ -168,9 +168,8 @@
 
 /obj/machinery/vending/RefreshParts()         //Better would be to make constructable child
 	if(component_parts)
-		build_inventory()
```
Comment (Fox-McCloud) [inline code/game/machinery/vending.dm]: Thatttt explains a lot.

**Comment by Tastyfish** [comment]
@KasparoVy Rebuilding it will still reset the machine to the default inventory, since building the machine brings everything to the default values. Something would have to be snowflaked into machine construction probably for it to do otherwise, unfortunately.

**Comment by KasparoVy** [comment]
Alright, I was just curious. Thanks for the fast response

**Pull request merged**